swift/apple/FirezoneNetworkExtension/CallbackHandler.swift

```diff
@@ -48,8 +48,11 @@ public class CallbackHandler {
           IPv6 routes: \(routeListv6.toString())
       """)
 
-    let dnsData = dnsAddresses.toString().data(using: .utf8)!
-    let dnsArray = try! JSONDecoder().decode([String].self, from: dnsData)
+    guard let dnsData = dnsAddresses.toString().data(using: .utf8),
+          let dnsArray = try? JSONDecoder().decode([String].self, from: dnsData)
+    else {
+      fatalError("Should be able to decode DNS Addresses from connlib")
```

Comment:
It would be nice if we weren't using JSON encoding here.

Reply:
Will come in #3959